### DISCUSSION TITLE
Allow skin elements to be placed behind game objects

### DISCUSSION THREAD

**Discussion by proxychns** [ppy/osu#30901] [Ideas]
Would allow us to do things like having a fully opaque play area image while keeping a low background dim.
Maybe a new HUD layer or even as a checkbox on the right click menu to keep the selected sprite/element behind the game objects.

![osu98675](https://github.com/user-attachments/assets/f9df3c4c-64a9-4710-ba5e-3a2c23fea263)

**Comment by yume-chan** [comment]
I want to move the HUD to the center, so I don't need to move my eyes to the corners to see them.

But now they cover the game objects. It's not unplayable, in fact, I even achieved a new highest PP with this setup.

---

I'm new to the code base, but I think adding another layer behind the play field needs either large refactoring or duplicating many code.

So I just swapped the game field and whole HUD to get an idea of how it would look. (Obviously this doesn't really work as the pause menu is now also below the game objects and unclickable). And I like it:

https://github.com/user-attachments/assets/db5d76a5-8a62-4610-ae0f-7efb028b90d6

**Reply by bdach** [reply]
I'm sorry but this video actively hurts my head. how are you even aiming for anything with the hud elements doing all sorts of animating behind the playfield

**Reply by normalid-awa** [reply]
I did actually place a gaint hit error meter in the centre of the screen😂 so if it was implemented, it would helps

**Reply by peppy** [reply]
We'll definitely add layers like this in the future. Right now, we're holding off so we can first do a full pass on the skinning system to make sure it's providing everything we need. This is to avoid migration complexities.

**Reply by Kuina-chan** [reply]
Bumping this, I love this idea
